resolwe/flow/managers/listener.py

```diff
@@ -1153,30 +1157,56 @@ async def _process_command(
         self, peer_identity: PeerIdentity, received_message: Message
     ):
         """Process command."""
-        response = await self._get_response(peer_identity, received_message)
-        try:
-            assert received_message.sent_timestamp is not None
-            response_time = time.time() - received_message.sent_timestamp
-            self.logger.debug(__("Response time: {}", response_time))
-            await self.communicator.send_response(response, peer_identity)
-
-        except RuntimeError:
-            # When unable to send message just log the error. Peer will be
-            # removed by the watchdog later on.
-            self.logger.exception(
-                "Protocol: error sending response to {received_message}."
+        if self._parallel_commands_semaphore is None:
```

Comment:
Is there a reason why you initialize semaphore here and not in the `__init__`?

Reply:
When semaphore is created it requires running event loop. If there is none, one is created.

If not being careful it can quickly happen that the semaphore is bound to a different event loop than the one that is used in listener. See 

https://stackoverflow.com/questions/55918048/asyncio-semaphore-runtimeerror-task-got-future-attached-to-a-different-loop

I just thought it is nicer to create semaphore when it is needed and avoid troubles with different event loops no matter how listener was initialized.

Reply:
Ok, thanks for the explanation.